File: python/src/common/Soup.py

# Soup- A helper class

class Soup:
	def __init__(self) :
		raise Exception("Can't instantiate abstract class, Soup")

	def __str__(self): 
		retval = [self.soupName, 'Ingredients: ']
		for i in self.soupIngredients:
			retval.append(i)
		return ', '.join(retval)

File: ruby/src/common/Soup.rb

# Soup- A helper class

class Soup
	attr_reader :soupName
	attr_writer :soupName
	
	def initialize()
		raise Exception("Can't instantiate abstract class, Soup")
	end

	def getSoupName
		@soupName
	end

	def to_s()
		retval = [@soupName, 'Ingredients: ']
		for i in @soupIngredients
			retval.push(i)
		end
		return retval.join(', ');
	end
end


